src/lib/stickers.ts

```diff
@@ -57,78 +56,63 @@ const stickerImageCache = LocalForage.createInstance({
 
 // ----- Functions -------------------------------------------------------------
 
-async function warmCachesIfNecessary() {
-  if (stickerPackListCache.length !== 0) {
-    return;
-  }
-
-  const res = await axios({
-    method: 'GET',
-    url: 'stickerData.json'
-  });
-
-  stickerPackListCache = res.data as Array<StickerPack>;
-
-  // Warm sticker manifest map.
-  stickerPackCache = R.reduce((result, value) => {
-    result.set(value.meta.id, value.manifest);
-    return result;
-  }, new Map<string, StickerPackManifest>(), stickerPackListCache);
-}
-
 /**
- * Loads and transforms stickers.json, which is the source of truth regarding
- * all sticker packs included in the application.
+ * Resolves with a list of StickerPackPartial objects.
  */
-export async function getStickerPackList(): Promise<Array<StickerPackMetadata>> {
-  await warmCachesIfNecessary();
-  return stickerPackListCache;
-}
-
+export async function getStickerPackDirectory(): Promise<Array<StickerPackPartial>> {
+  if (!stickerPackDirectoryPromise) {
+    stickerPackDirectoryPromise = new Promise(async (resolve, reject) => {
+      const res = await axios.request<Array<StickerPackPartial>>({
+        method: 'GET',
+        url: 'stickerData.json'
+      });
 
-/**
- * Provided a key and an encrypted manifest from the Signal API, resolves with a
- * decrypted and parsed manifest.
- */
-export async function parseManifest(key: string, rawManifest: any): Promise<StickerPackManifest> {
-  try {
-    const manifest = await decryptManifest(key, rawManifest);
-    const PackMessage = protobufClient.lookupType('Pack');
-    const manifestData = new Uint8Array(manifest, 0, manifest.byteLength);
-    return PackMessage.decode(manifestData) as unknown as StickerPackManifest;
-  } catch (err) {
-    throw new ErrorWithCode(err.code || 'MANIFEST_PARSE', `[parseManifest] ${err.message}`);
+      resolve(res.data);
+    });
   }
+
+  return stickerPackDirectoryPromise;
 }
 
 
 /**
- * Provided a sticker pack ID and key, queries the Signal API and resolves with
- * a parsed manifest.
+ * Provided a sticker pack ID and optional key, queries the Signal API and
+ * resolves with 'full' StickerPack object.
  */
-export async function getStickerPack(id: string, key: string, fetchStickersIfNecessary: boolean | false): Promise<StickerPackManifest> {
+export async function getStickerPack(id: string, key?: string): Promise<StickerPack> {
```

Comment:
Great catch, fixed in 6b204c76b32100cb7e455d81a9f8d846e4d6db27.

Turns out by removing the convenience of allowing callee's to pass `'cover'` as a sticker ID, we no longer needed these functions to get manifests to perform that lookup, we can just get the sticker (image data) directly. And by virtue of having the cover from the manifest in `stickerData.json` (re-added) we always know the ID of the cover now anyway, so that convenience was superfluous.